OpenRA.Mods.Common/Traits/Explodes.cs

```diff
@@ -41,7 +41,7 @@ public void Killed(Actor self, AttackInfo e)
 			if (self.World.SharedRandom.Next(100) > explodesInfo.Chance)
 				return;
 
-			if (explodesInfo.DeathType != null && e.Warhead != null && !explodesInfo.DeathType.Contains(e.Warhead.DeathType))
+			if (explodesInfo.DeathType != null && e.Warhead != null && !explodesInfo.DeathType.Any(x => e.Warhead.DamageTypes.Contains(x)))
```

Comment:
I know. I'm just trying to avoid pretty-but-inefficient code.
But OK, these will never contain enough elements for it to matter, so I won't argue here (it does set a bad example though).